### PR TITLE
#72 migrate forgot pass to angular

### PR DESCRIPTION
**Pull request by Sheptytskyid**

### CODE REVIEW

File: src/main/java/net/greatstart/controllers/PasswordResetController.java [hunk 1]
```diff
@@ -1,117 +1,90 @@
...
 
     @GetMapping(value = "/user/validateToken")
     public ModelAndView validatePassToken(Locale locale, @RequestParam("id") long id, @RequestParam("token") String token) {
         String result = securityService.validatePasswordResetToken(id, token, locale);
         ModelAndView model = new ModelAndView();
         if (result != null) {
-            model.addObject(MESSAGE, messages.getMessage("token.error", null, locale) + result);
-            model.setViewName("login/login");
+            model.setViewName("redirect:/");
+            //TODO: implement displaying errors, maybe in some generic error page
```
Comment (Sheptytskyid) [inline src/main/java/net/greatstart/controllers/PasswordResetController.java]: INFO Complete the task associated to this TODO comment. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AS1135)

**Comment by Sheptytskyid** [comment]
SonarQube analysis reported 1 issue
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

**Comment by holkwen** [comment]
Я думаю, модальные окна можно закрывать. Например, при показе окна восстановления пароля окно логина становится бессмысленным, его можно закрыть. После отсылки линка можно выводить кнопку "ОК" для закрытия окна восстановления.

А так вроде всё нормально.

**Comment by Sheptytskyid** [comment]
@holkwen Done!

File: src/main/java/net/greatstart/controllers/PasswordResetController.java [hunk 2]
```diff
@@ -1,117 +1,79 @@
...
-
-    @PostMapping(value = "/user/resetPassword")
-    public ModelAndView processEmail(HttpServletRequest request, @Valid @ModelAttribute("user") DtoUser dtoUser, Errors
-        errors) {
-        ModelAndView model = new ModelAndView("user/forgotPassword");
-        if (errors.hasErrors()) {
-            return model;
-        }
-        User user = userService.getUserByEmail(dtoUser.getEmail());
+    public ResponseEntity<String[]> processEmail(HttpServletRequest request, @RequestParam("email") String email) {
```
Comment (zozich) [inline src/main/java/net/greatstart/controllers/PasswordResetController.java]: 1. No need to indicate param name if it has the same name as imported variable
2. Why do you return String[]?

File: src/main/java/net/greatstart/controllers/PasswordResetController.java [hunk 3]
```diff
@@ -1,117 +1,79 @@
...
-            model.setViewName("user/updatePassword");
-        } else {
-            User user = (User) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
-            userService.changeUserPassword(user, passwordEncoder.encode(dtoUser.getPassword()));
-            securityService.expireToken(user);
-            model.setViewName("login/login");
-            model.addObject(MESSAGE, messages.getMessage("message.resetPassword.success", null, locale));
-        }
-        return model;
+    @ExceptionHandler(ServiceException.class)
```
Comment (zozich) [inline src/main/java/net/greatstart/controllers/PasswordResetController.java]: You should have one generic exception handler. There is a task for that

File: src/test.java.net.greatstart.model/net/greatstart/model/ContactTest.java [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    private Contact contact;
+
+    @Before
+    public void setUp() {
+        contact = new Contact("StreetTest", "123456");
+    }
+
+    @Test
+    public void addressShouldEqual() {
+        assertEquals("StreetTest", contact.getAddress());
```
Comment (zozich) [inline src/test.java.net.greatstart.model/net/greatstart/model/ContactTest.java]: WTF?

File: src/test/java/net/greatstart/controllers/PasswordResetControllerTest.java [hunk 5]
```diff
@@ -61,134 +46,61 @@
...
-        assertTrue(controller.processEmail(request, dtoUser, new BindException(dtoUser, "user"))
-                            .getModel()
-                            .get("message")
-                            .equals(EMAIL_ERROR));
-
+    public void processNonExistingEmailShouldReturnBadRequestStatus() throws Exception {
+        mvc.perform(get("/user/resetPassword").param("email", VALID_EMAIL)).andExpect(status().isBadRequest());
     }
 
     @Test(timeout = 2000)
```
Comment (zozich) [inline src/test/java/net/greatstart/controllers/PasswordResetControllerTest.java]: Why do you have timeouts here?

File: src/main/java/net/greatstart/controllers/PasswordResetController.java [hunk 6]
```diff
@@ -1,117 +1,79 @@
...
+    public ResponseEntity<String[]> processEmail(HttpServletRequest request, @RequestParam("email") String email) {
+        User user = userService.getUserByEmail(email);
         if (user == null) {
-            model.addObject(MESSAGE, messages.getMessage("user.notFound", null, request.getLocale()));
-            return model;
+            throw new ServiceException(messages.getMessage("user.notFound", null, request.getLocale()));
         }
         String passwordResetToken = securityService.createPasswordResetToken(user).getToken();
-        String url = request.getHeader("origin");
+        String url = request.getHeader("referer");
```
Comment (zozich) [inline src/main/java/net/greatstart/controllers/PasswordResetController.java]: There should be no business logic in controller

**Review by zozich — APPROVED** [review]
This should be reworked. Approving only because of lack of time

File: src/main/resources/static/views/main/PasswordResetController.js [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+
+        $scope.closeFpModal = function () {
+            $scope.forgotPassModal.dismiss();
+        };
+
+        $scope.submitEmail = function () {
+            $scope.submitted = true;
+            $scope.message = null;
+            $scope.error = null;
+            var data = passwordResetService.get({email:$scope.email}, function(){
```
Comment (Sheptytskyid) [inline src/main/resources/static/views/main/PasswordResetController.js]: MAJOR Remove this useless assignment to local variable "data" [rule](https://sonarqube.com/coding_rules#rule_key=javascript%3AS1854)
MINOR Remove the declaration of the unused 'data' variable. [rule](https://sonarqube.com/coding_rules#rule_key=javascript%3AUnusedVariable)

**Comment by Sheptytskyid** [comment]
SonarQube analysis reported 2 issues
* ![MAJOR][MAJOR] 1 major
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**